File: yaf-gadomain/src/main/java/ar/edu/ungs/yamiko/ga/domain/impl/BasicIndividual.java
package ar.edu.ungs.yamiko.ga.domain.impl;

import java.io.Serializable;

import ar.edu.ungs.yamiko.ga.domain.Genotype;
import ar.edu.ungs.yamiko.ga.domain.Individual;
import ar.edu.ungs.yamiko.ga.domain.Phenotype;

/**
 * Implementación Básica de Individuo.
 * @author ricardo
 *
 * @param <T>
 */
public class BasicIndividual<T> implements Individual<T>,Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 142698255700324496L;
	private Double fitness;
	private Genotype<T> genotype;
	private Phenotype phenotype;
	private long id;
	
	public BasicIndividual() {

	}
	
	public Genotype<T> getGenotype() {
		return genotype;
	}
	
	public Phenotype getPhenotype() {
		return phenotype;
	}

	public Double getFitness() {
		return fitness;
	}

	public void setFitness(Double fitness) {
		this.fitness = fitness;
	}

	public void setGenotype(Genotype<T> genotype) {
		this.genotype = genotype;
	}

	public void setPhenotype(Phenotype phenotype) {
		this.phenotype = phenotype;
	}

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + (int) (id ^ (id >>> 32));
		return result;
	}

	@SuppressWarnings("rawtypes")
	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		BasicIndividual other = (BasicIndividual) obj;
		if (id != other.id)
			return false;
		return true;
	}

	
	
	
	
}

File: yaf-gadomain-scala/src/main/scala/ar/edu/ungs/yamiko/ga/domain/impl/BasicIndividual.scala
package ar.edu.ungs.yamiko.ga.domain.impl

import ar.edu.ungs.yamiko.ga.domain.Gene
import ar.edu.ungs.yamiko.ga.domain.Genome
import ar.edu.ungs.yamiko.ga.domain.Ribosome
import ar.edu.ungs.yamiko.ga.domain.Genotype
import ar.edu.ungs.yamiko.ga.domain.Individual
import ar.edu.ungs.yamiko.ga.domain.Phenotype
import ar.edu.ungs.yamiko.ga.domain.Chromosome
import ar.edu.ungs.yamiko.ga.domain.Phenotype
import ar.edu.ungs.yamiko.ga.toolkit.IntArrayHelper



@SerialVersionUID(1179L)
class BasicIndividual[T](genotype:Genotype[T],id:Int) extends Individual[T]{
  
	private var phenotype:Phenotype[T] =null
	private var fitness:Double=0d
	private var intAttachment:List[Int]=null
	
	override def getGenotype()=genotype
	override def getPhenotype():Phenotype[T]={return phenotype}
	override def setPhenotype(phe:Phenotype[T])={phenotype=phe}
	override def getFitness():Double={return fitness}
	override def setFitness(v:Double)={fitness=v}
  override def getId():Int=id
	override def getIntAttachment():List[Int]=intAttachment
	override def setIntAttachment(a:List[Int])=intAttachment=a

	override def toString = "BasicIndividual [id=" + id + ", fitness=" + fitness + "]"
	override def toStringRepresentation = if  (this.getGenotype().getChromosomes()(0).getFullRawRepresentation().isInstanceOf[Array[Int]]) IntArrayHelper.toStringIntArray(this.getGenotype().getChromosomes()(0).getFullRawRepresentation().asInstanceOf[Array[Int]])  else this.getGenotype().getChromosomes()(0).getFullRawRepresentation().toString()  

		
	
  def canEqual(a: Any) = a.isInstanceOf[BasicIndividual[T]]

  override def equals(that: Any): Boolean =
    that match {
      case that: BasicIndividual[T] => 
        that.canEqual(this) && this.hashCode == that.hashCode
      case _ => false
  }

  override def hashCode:Int = super.hashCode + id.hashCode()
  
  


	
	
  
  
  
}